crates/client/src/storage.rs

```diff
@@ -34,6 +35,14 @@ pub trait RegistryStorage: Send + Sync {
     /// Stores most recent checkpoint
     async fn store_checkpoint(&self, checkpoint: SerdeEnvelope<MapCheckpoint>) -> Result<()>;
 
+    /// Loads the operator information from the storage.
+    ///
+    /// Returns `Ok(None)` if the information is not present.
+    async fn load_operator(&self) -> Result<Option<OperatorRecord>>;
```

Comment:
This doesn't seem to be called. Additionally, I'd expect this to return `OperatorInfo`.